### PR TITLE
Mongo services (part one)

### PR DESCRIPTION
**Pull request by Piokor**
Created mongo api service for comunication with mongodb. It enables adding/getting/updating/deleting mongo documents. It also manages difference in type used for ids - mongo uses ObjectId, project models use string.

Created mongo services for:
- channel
- registered channel
- registered data
- observable information
- recording

Additionally added GenericMongoServiceMixin that hold generic implementation of mongo service functionalities - it greatly reduced code repeatability.

### CODE REVIEW

File: grisera_api/channel/channel_service_mongodb.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def get_channel(
+        self, channel_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to mongo api to get given channel with related models. This method uses mixin get implementation.
+
+        Args:
+            channel_id (int): Id of channel
+            depth (int): this attribute specifies how many models will be traversed to create the response.
+                         for depth=0, only no further models will be travesed.
```
Comment (marcel2012) [inline grisera_api/channel/channel_service_mongodb.py:53]:
```suggestion
                         for depth=0, only no further models will be traversed.
```

File: grisera_api/channel/channel_service_mongodb.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from channel.channel_model import ChannelIn, ChannelOut, ChannelsOut, BasicChannelOut
+from mongo_service.collection_mapping import Collections
+from mongo_service.service_mixins import GenericMongoServiceMixin
+
+
+class ChannelServiceMongoDB(ChannelService, GenericMongoServiceMixin):
+    """
+    Object to handle logic of channel requests
+
+    Attributes:
```
Comment (marcel2012) [inline grisera_api/channel/channel_service_mongodb.py:12]: missing `model_out_class` attribute

File: grisera_api/mongo_service/collection_mapping.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+from registered_data.registered_data_model import RegisteredDataIn
+from scenario.scenario_model import ScenarioIn
+from time_series.time_series_model import (
+    TimeSeriesPropertyIn,
+    TimeSeriesRelationIn,
+)
+from enum import Enum
+
+"""
+This module provides enum with collecion names in mongodb to help avoiding
```
Comment (marcel2012) [inline grisera_api/mongo_service/collection_mapping.py:47]:
```suggestion
This module provides enum with collection names in mongodb to help avoiding 
```

File: grisera_api/mongo_service/collection_mapping.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    TimeSeriesPropertyIn,
+    TimeSeriesRelationIn,
+)
+from enum import Enum
+
+"""
+This module provides enum with collecion names in mongodb to help avoiding 
+possible errors with typos in collection names strings.
+
+It also provides mapping of model classes to collection names. It is usefull
```
Comment (marcel2012) [inline grisera_api/mongo_service/collection_mapping.py:50]:
```suggestion
It also provides mapping of model classes to collection names. It is useful 
```

File: grisera_api/mongo_service/service_mixins.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+from pydantic import BaseModel
+from mongo_service.collection_mapping import get_collection_name
+from models.not_found_model import NotFoundByIdModel
+from mongo_service import MongoApiService
+
+
+class GenericMongoServiceMixin:
+
+    """
+    This mixin defines implementation of basic mongo services methods. It requires the sublass to implement:
```
Comment (marcel2012) [inline grisera_api/mongo_service/service_mixins.py:13]:
```suggestion
    This mixin defines implementation of basic mongo services methods. It requires the subclass to implement:
```

File: grisera_api/mongo_service/service_mixins.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        model_out_class field - out model class of services model. Based on this atribute mongo collection is
```
Comment (marcel2012) [inline grisera_api/mongo_service/service_mixins.py:14]:
```suggestion
        model_out_class field - out model class of services model. Based on this attribute mongo collection is
```

File: grisera_api/mongo_service/service_mixins.py [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        return results_dict
+
+    def get_single_dict(
+        self, id: Union[str, int], depth: int = 0, source: str = "", *args, **kwargs
+    ):
+        """
+        Generic method for getting a single document in dict form from mongo api.
+
+        Args:
+            query: Query to mongo api. Empty by default.
```
Comment (marcel2012) [inline grisera_api/mongo_service/service_mixins.py:67]: wrong argument name

File: grisera_api/mongo_service/service_mixins.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        if type(result) is NotFoundByIdModel:
+            return result
+        return out_class(**result)
+
+    def update(self, id: Union[str, int], updated_object: BaseModel):
+        """
+        Generic method for sending request to mongo api to update single document
+
+        Args:
+            id: Id of docuemnt to be updated.
```
Comment (marcel2012) [inline grisera_api/mongo_service/service_mixins.py:111]:
```suggestion
            id: Id of document to be updated.
```

File: grisera_api/mongo_service/service_mixins.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+        self.mongo_api_service.update_document(id, updated_object)
+
+        return self.get_single(id)
+
+    def delete(self, id: Union[str, int]):
+        """
+        Generic method for delete request to mongo api
+
+        Args:
+            id: Id of docuemnt to be deleted.
```
Comment (marcel2012) [inline grisera_api/mongo_service/service_mixins.py:131]:
```suggestion
            id: Id of document to be deleted.
```

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+
+class ObservableInformationServiceMongoDB(
+    ObservableInformationService, GenericMongoServiceMixin
+):
+    """
+    Object to handle logic of observable information requests Observable information documents
+    are embedded within recording documents.
+
+    Attributes:
+    graph_api_service (GraphApiService): Service used to communicate with Graph API
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:25]: why graph api?

File: grisera_api/recording/recording_service_mongodb.py [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+from mongo_service import MongoApiService
+
+
+class RecordingServiceMongoDB(RecordingService, GenericMongoServiceMixin):
+    """
+    Object to handle logic of recording requests. There are also observable information methods,
+    as observable information documents are embedded within recording documents.
+
+    Attributes:
+    graph_api_service (GraphApiService): Service used to communicate with Graph API
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:32]: why graph api?

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+        self.recording_service = None
+        self.life_activity_service = None
+        self.modality_service = None
+        self.time_series_service = None
+
+    def save_observable_information(
+        self, observable_information: ObservableInformationIn
+    ):
+        """
+        Send request to graph api to create new observable information. Saving is performed by
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:43]: graph -> mongo

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+                    source,
+                    recording_result,
+                )
+            result += observable_informations
+
+        return result
+
+    def get_observable_informations(self):
+        """
+        Send request to graph api to get all observable informations.
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:127]: graph -> mongo

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 14]
```diff
@@ -0,0 +1,282 @@
...
+        Args:
+            observable_information_id (int): Id of observable information
+        Returns:
+            Result of request as observable information object
+        """
+        return self.get_single(observable_information_id, depth, source)
+
+    def delete_observable_information(self, observable_information_id: Union[str, int]):
+        """
+        Send request to graph api to delete given observable information. Removal is performed by recording service,
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:200]: graph -> mongo

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+        return ObservableInformationOut(**result)
+
+    def get_observable_information(
+        self,
+        observable_information_id: Union[str, int],
+        depth: int = 0,
+        source: str = "",
+    ):
+        """
+        Send request to graph api to get given observable information
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:190]: graph -> mongo

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+            observable_information
+        )
+
+    def update_observable_information_relationships(
+        self,
+        observable_information_id: Union[str, int],
+        observable_information: BasicObservableInformationOut,
+    ):
+        """
+        Send request to graph api to update given observable information
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:225]: graph -> mongo

File: grisera_api/registered_data/registered_data_service_mongodb.py [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+        """
+        result_dict = self.get_multiple()
+        results = [BasicRegisteredDataOut(**result) for result in result_dict]
+        return RegisteredDataNodesOut(registered_data_nodes=results)
+
+    def get_registered_data(
+        self, registered_data_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to graph api to get given registered data with related models. This method uses mixin get implementation.
```
Comment (marcel2012) [inline grisera_api/registered_data/registered_data_service_mongodb.py:55]: graph -> mongo

File: grisera_api/recording/recording_service_mongodb.py [hunk 18]
```diff
@@ -0,0 +1,331 @@
...
+        ):
+            return RecordingOut(
+                errors={"errors": "given registered channel does not exist"}
+            )
+
+        return self.create(recording)
+
+    def get_recordings(self, query: dict = {}):
+        """
+        Send request to graph api to get recordings
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:79]: graph -> mongo

File: grisera_api/recording/recording_service_mongodb.py [hunk 19]
```diff
@@ -0,0 +1,331 @@
...
+        Returns:
+            Result of request as recording object
+        """
+        return self.delete(recording_id)
+
+    def update_recording(
+        self, recording_id: Union[str, int], recording: RecordingPropertyIn
+    ):
+        """
+        Send request to graph api to update given participant state
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:123]: graph -> mongo

File: grisera_api/recording/recording_service_mongodb.py [hunk 20]
```diff
@@ -0,0 +1,331 @@
...
+        Returns:
+            Result of request as participant state object
+        """
+        return self.update(recording_id, recording)
+
+    def update_recording_relationships(
+        self, recording_id: Union[str, int], recording: RecordingIn
+    ):
+        """
+        Send request to graph api to update given recording
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:138]: graph -> mongo

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 21]
```diff
@@ -0,0 +1,282 @@
...
+        return ObservableInformationsOut(observable_informations=results)
+
+    def get_single_dict(
+        self, id: Union[str, int], depth: int = 0, source: str = "", *args, **kwargs
+    ):
+        """
+        Get observable information dict. Observable information is fetched from its
+        recording.
+        """
+        observable_information_objectid = ObjectId(id)
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:143]:
```suggestion
        observable_information_object_id = ObjectId(id)
```

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 22]
```diff
@@ -0,0 +1,282 @@
...
+    def get_observable_information(
+        self,
+        observable_information_id: Union[str, int],
+        depth: int = 0,
+        source: str = "",
+    ):
+        """
+        Send request to graph api to get given observable information
+        Args:
+            observable_information_id (int): Id of observable information
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:192]: missing arguments

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 23]
```diff
@@ -0,0 +1,282 @@
...
+        Returns:
+            Result of request as observable information object
+        """
+        return self.get_single(observable_information_id, depth, source)
+
+    def delete_observable_information(self, observable_information_id: Union[str, int]):
+        """
+        Send request to graph api to delete given observable information. Removal is performed by recording service,
+        as observable information is embeded within recording
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:201]:
```suggestion
        as observable information is embedded within recording
```

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 24]
```diff
@@ -0,0 +1,282 @@
...
+
+    def update_observable_information_relationships(
+        self,
+        observable_information_id: Union[str, int],
+        observable_information: BasicObservableInformationOut,
+    ):
+        """
+        Send request to graph api to update given observable information
+        Args:
+            observable_information_id (int): Id of observable information
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:227]: wrong type -> int | string

File: grisera_api/recording/recording_service_mongodb.py [hunk 25]
```diff
@@ -0,0 +1,331 @@
...
+        return self.delete(recording_id)
+
+    def update_recording(
+        self, recording_id: Union[str, int], recording: RecordingPropertyIn
+    ):
+        """
+        Send request to graph api to update given participant state
+
+        Args:
+            recording_id (int): Id of participant state
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:126]: wrong type -> int | string

File: grisera_api/channel/channel_service_mongodb.py [hunk 26]
```diff
@@ -0,0 +1,69 @@
...
+        return ChannelsOut(channels=results)
+
+    def get_channel(
+        self, channel_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to mongo api to get given channel with related models. This method uses mixin get implementation.
+
+        Args:
+            channel_id (int): Id of channel
```
Comment (marcel2012) [inline grisera_api/channel/channel_service_mongodb.py:51]: wrong type -> int | string

File: grisera_api/mongo_service/service_mixins.py [hunk 27]
```diff
@@ -0,0 +1,145 @@
...
+        return result_dict
+
+    def get_single(
+        self, id: Union[str, int], depth: int = 0, source: str = "", *args, **kwargs
+    ):
+        """
+        Generic method for getting a single documents from mongo api.
+
+        Args:
+            query: Query to mongo api. Empty by default.
```
Comment (marcel2012) [inline grisera_api/mongo_service/service_mixins.py:93]: wrong argument name

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 28]
```diff
@@ -0,0 +1,282 @@
...
+            Result of request as observable information object
+        """
+        return self.get_single(observable_information_id, depth, source)
+
+    def delete_observable_information(self, observable_information_id: Union[str, int]):
+        """
+        Send request to graph api to delete given observable information. Removal is performed by recording service,
+        as observable information is embeded within recording
+        Args:
+            observable_information_id (int): Id of observable information
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:203]: wrong type -> int | string

File: grisera_api/recording/recording_service_mongodb.py [hunk 29]
```diff
@@ -0,0 +1,331 @@
...
+        return RecordingsOut(registered_channels=results)
+
+    def get_recording(
+        self, recording_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to mongo api to get given recording. This method uses mixin get implementation.
+
+        Args:
+            recording_id (int): Id of registered channel
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:94]: wrong type -> int | string

File: grisera_api/recording/recording_service_mongodb.py [hunk 30]
```diff
@@ -0,0 +1,331 @@
...
+            Result of request as registered channel object
+        """
+        return self.get_single(recording_id, depth, source)
+
+    def delete_recording(self, recording_id: Union[str, int]):
+        """
+        Send request to mongo api to delete given recording
+
+        Args:
+            recording_id (int): Id of recording
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:112]: wrong type -> int | string

File: grisera_api/recording/recording_service_mongodb.py [hunk 31]
```diff
@@ -0,0 +1,331 @@
...
+        return self.update(recording_id, recording)
+
+    def update_recording_relationships(
+        self, recording_id: Union[str, int], recording: RecordingIn
+    ):
+        """
+        Send request to graph api to update given recording
+
+        Args:
+            recording_id (int): Id of recording
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:141]: wrong type -> int | string

File: grisera_api/recording/recording_service_mongodb.py [hunk 32]
```diff
@@ -0,0 +1,331 @@
...
+    def update_observable_information(
+        self,
+        observable_information_id: Union[int, str],
+        observable_information_dict: dict,
+    ):
+        """
+        Edit observable information in recording. Obervable information is embeded in related recording.
+
+        Args:
+            observable_information (BasicObservableInformationOut): new version of observable information
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:212]: wrong arguments

File: grisera_api/registered_data/registered_data_service_mongodb.py [hunk 33]
```diff
@@ -0,0 +1,103 @@
...
+            Result of request as registered data object
+        """
+        return self.update(registered_data_id, registered_data)
+
+    def delete_registered_data(self, registered_data_id: Union[str, int]):
+        """
+        Send request to mongo api to delete given registered data. This method uses mixin delete implementation.
+
+        Args:
+        registered_data_id (int): Id of registered data
```
Comment (marcel2012) [inline grisera_api/registered_data/registered_data_service_mongodb.py:88]: wrong type -> int | string

File: grisera_api/registered_data/registered_data_service_mongodb.py [hunk 34]
```diff
@@ -0,0 +1,103 @@
...
+        return self.get_single(registered_data_id, depth, source)
+
+    def update_registered_data(
+        self, registered_data_id: Union[str, int], registered_data: RegisteredDataIn
+    ):
+        """
+        Send request to mongo api to update given registered data. This method uses mixin update implementation.
+
+        Args:
+        registered_data_id (int): Id of registered data
```
Comment (marcel2012) [inline grisera_api/registered_data/registered_data_service_mongodb.py:75]: wrong type -> int | string

File: grisera_api/registered_data/registered_data_service_mongodb.py [hunk 35]
```diff
@@ -0,0 +1,103 @@
...
+        return RegisteredDataNodesOut(registered_data_nodes=results)
+
+    def get_registered_data(
+        self, registered_data_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to graph api to get given registered data with related models. This method uses mixin get implementation.
+
+        Args:
+            registered_channel_id (int): Id of registered data
```
Comment (marcel2012) [inline grisera_api/registered_data/registered_data_service_mongodb.py:58]: wrong type -> int | string

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 36]
```diff
@@ -0,0 +1,282 @@
...
+    def update_observable_information_relationships(
+        self,
+        observable_information_id: Union[str, int],
+        observable_information: BasicObservableInformationOut,
+    ):
+        """
+        Send request to graph api to update given observable information
+        Args:
+            observable_information_id (int): Id of observable information
+            observable_information (ObservableInformationIn): Relationships to update
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:228]: wrong type -> BasicObservableInformationOut

File: grisera_api/observable_information/observable_information_service_mongodb.py [hunk 37]
```diff
@@ -0,0 +1,282 @@
...
+    ):
+        pass
+        # TODO add when time series service is ready
+
+    @staticmethod
+    def _get_recording_projection(query):
+        return {
+            Collections.OBSERVABLE_INFORMATION: {"$elemMatch": query},
+            "additional_properties": 1,
+            "participation_id": 1,
```
Comment (marcel2012) [inline grisera_api/observable_information/observable_information_service_mongodb.py:280]: why this ids are hardcoded? is it TODO?

File: grisera_api/recording/recording_service_mongodb.py [hunk 38]
```diff
@@ -0,0 +1,331 @@
...
+    def get_recording(
+        self, recording_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to mongo api to get given recording. This method uses mixin get implementation.
+
+        Args:
+            recording_id (int): Id of registered channel
+            depth (int): this attribute specifies how many models will be traversed to create the response.
+                         for depth=0, only no further models will be travesed.
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:96]:
```suggestion
                         for depth=0, only no further models will be traversed.
```

File: grisera_api/recording/recording_service_mongodb.py [hunk 39]
```diff
@@ -0,0 +1,331 @@
...
+                RecordingIn(participation_id=recording.participation_id),
+            )
+
+        return self.get_recording(recording_id)
+
+    def add_observable_information(
+        self, observable_information: ObservableInformationIn
+    ):
+        """
+        Add observable information to recording. Obervable information is embeded in related recording.
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:182]:
```suggestion
        Add observable information to recording. Observable information is embedded in related recording.
```

File: grisera_api/recording/recording_service_mongodb.py [hunk 40]
```diff
@@ -0,0 +1,331 @@
...
+        self.update(recording_id, RecordingOut(**recording))
+        return BasicObservableInformationOut(**observable_information_dict)
+
+    def update_observable_information(
+        self,
+        observable_information_id: Union[int, str],
+        observable_information_dict: dict,
+    ):
+        """
+        Edit observable information in recording. Obervable information is embeded in related recording.
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:209]:
```suggestion
        Edit observable information in recording. Observable information is embedded in related recording.
```

File: grisera_api/recording/recording_service_mongodb.py [hunk 41]
```diff
@@ -0,0 +1,331 @@
...
+        observable_informations = recording[Collections.OBSERVABLE_INFORMATION]
+        observable_informations[to_update_index] = observable_information_dict
+        self.update(recording_id, recording)
+        return observable_information_dict
+
+    def remove_observable_information(
+        self, observable_information: ObservableInformationOut
+    ):
+        """
+        Remove observable information from recording. Obervable information is embeded in related recording.
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:245]:
```suggestion
        Remove observable information from recording. Observable information is embedded in related recording.
```

File: grisera_api/recording/recording_service_mongodb.py [hunk 42]
```diff
@@ -0,0 +1,331 @@
...
+                channel_id=recording["participation_id"],
+                depth=depth - 1,
+                source=Collections.RECORDING,
+            )
+
+    def _add_related_observable_informations(
+        self, recording: dict, depth: int, source: str
+    ):
+        """
+        Oservable information is embeded within recording model
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:306]:
```suggestion
        Observable information is embedded within recording model
```

File: grisera_api/recording/recording_service_mongodb.py [hunk 43]
```diff
@@ -0,0 +1,331 @@
...
+            for oi in recording[Collections.OBSERVABLE_INFORMATION]:
+                self.observable_information_service._add_related_documents(
+                    oi, depth - 1, Collections.RECORDING, recording
+                )
+
+    def _get_observable_information_index_from_recording(
+        self, recording_dict: dict, observable_information_id: Union[str, int]
+    ):
+        """
+        Oservable information is embeded within recording model
```
Comment (marcel2012) [inline grisera_api/recording/recording_service_mongodb.py:321]:
```suggestion
        Observable information is embedded within recording model
```

File: grisera_api/registered_channel/registered_channel_service_mongodb.py [hunk 44]
```diff
@@ -0,0 +1,201 @@
...
+    def get_registered_channel(
+        self, registered_channel_id: int, depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to mongo api to get given registered channel. This method uses mixin get implementation.
+
+        Args:
+            registered_channel_id (int): Id of registered channel
+            depth (int): this attribute specifies how many models will be traversed to create the response.
+                         for depth=0, only no further models will be travesed.
```
Comment (marcel2012) [inline grisera_api/registered_channel/registered_channel_service_mongodb.py:89]:
```suggestion
                         for depth=0, only no further models will be traversed.
```

File: grisera_api/registered_channel/registered_channel_service_mongodb.py [hunk 45]
```diff
@@ -0,0 +1,201 @@
...
+        self,
+        registered_channel_id: int,
+        updated_registered_channel: RegisteredChannelIn,
+    ):
+        """
+        Send request to mongo api to update given registered channel
+
+        Args:
+            registered_channel_id (int): Id of registered channel
+            registered_channel (RegisteredChannelIn): Document to update
```
Comment (marcel2012) [inline grisera_api/registered_channel/registered_channel_service_mongodb.py:109]: wrong argument name

File: grisera_api/registered_data/registered_data_service_mongodb.py [hunk 46]
```diff
@@ -0,0 +1,103 @@
...
+            Result of request as registered data object
+        """
+        return self.create(registered_data)
+
+    def get_registered_data_nodes(self):
+        """
+        Send request to mongo api to get ass registered data nodes. This method uses mixin get implementation.
+
+        Args:
+            query (dict): Query for mongo request.
```
Comment (marcel2012) [inline grisera_api/registered_data/registered_data_service_mongodb.py:42]: wrong argument

File: grisera_api/registered_data/registered_data_service_mongodb.py [hunk 47]
```diff
@@ -0,0 +1,103 @@
...
+    def get_registered_data(
+        self, registered_data_id: Union[str, int], depth: int = 0, source: str = ""
+    ):
+        """
+        Send request to graph api to get given registered data with related models. This method uses mixin get implementation.
+
+        Args:
+            registered_channel_id (int): Id of registered data
+            depth (int): this attribute specifies how many models will be traversed to create the response.
+                         for depth=0, only no further models will be travesed.
```
Comment (marcel2012) [inline grisera_api/registered_data/registered_data_service_mongodb.py:60]:
```suggestion
                         for depth=0, only no further models will be traversed.
```

File: grisera_api/tests/tests_mongodb/test_observable_information.py [hunk 48]
```diff
@@ -0,0 +1,67 @@
...
+        if not save:
+            return recording
+        service = Services().recording_service()
+        return service.save_recording(recording)
+
+    @mongomock.patch(servers=((mongo_api_host, mongo_api_port),))
+    def test_create(self):
+        service = Services().observable_information_service()
+        recording = self.generate_recording(save=True)
+        observable_infromation = ObservableInformationIn(recording_id=recording.id)
```
Comment (marcel2012) [inline grisera_api/tests/tests_mongodb/test_observable_information.py:25]: please replace all `infromation` to `information` in this file

File: grisera_api/tests/tests_mongodb/test_recording.py [hunk 49]
```diff
@@ -0,0 +1,84 @@
+import mongomock
+
+from observable_information.observable_information_model import ObservableInformationIn
+from registered_channel.registered_channel_model import RegisteredChannelIn
+from tests.tests_mongodb.utils import MongoTestCase
+from models.not_found_model import NotFoundByIdModel
+from services import Services
+from registered_data.registered_data_model import RegisteredDataIn
```
Comment (marcel2012) [inline grisera_api/tests/tests_mongodb/test_recording.py:8]: unused import

File: grisera_api/tests/tests_mongodb/test_registered_channel.py [hunk 50]
```diff
@@ -0,0 +1,104 @@
...
+
+        service.delete_registered_channel(created_rc.id)
+        get_result = service.get_registered_channel(created_rc.id)
+        self.assertTrue(type(get_result) is NotFoundByIdModel)
+
+    @mongomock.patch(servers=((mongo_api_host, mongo_api_port),))
+    def test_traverse_one(self):
+        service = Services().registered_channel_service()
+        registered_data_service = Services().registered_data_service()
+        channel_service = Services().channel_service()
```
Comment (marcel2012) [inline grisera_api/tests/tests_mongodb/test_registered_channel.py:81]: unused variables

**Review by marcel2012 — CHANGES_REQUESTED** [review]

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 37 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py:280]: 1s here are not IDs, in mongo projection "1" is the argument that specifies that field should be included in result.
[https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/](https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/)

File: grisera_api/channel/channel_service_mongodb.py (hunk 1 above)
Comment (Piokor) [reply grisera_api/channel/channel_service_mongodb.py]: fixed

File: grisera_api/channel/channel_service_mongodb.py (hunk 2 above)
Comment (Piokor) [reply grisera_api/channel/channel_service_mongodb.py:12]: fixed

File: grisera_api/mongo_service/collection_mapping.py (hunk 3 above)
Comment (Piokor) [reply grisera_api/mongo_service/collection_mapping.py]: fixed

File: grisera_api/mongo_service/collection_mapping.py (hunk 4 above)
Comment (Piokor) [reply grisera_api/mongo_service/collection_mapping.py]: fixed

File: grisera_api/mongo_service/service_mixins.py (hunk 5 above)
Comment (Piokor) [reply grisera_api/mongo_service/service_mixins.py]: fixed

File: grisera_api/mongo_service/service_mixins.py (hunk 6 above)
Comment (Piokor) [reply grisera_api/mongo_service/service_mixins.py]: fixed

File: grisera_api/mongo_service/service_mixins.py (hunk 9 above)
Comment (Piokor) [reply grisera_api/mongo_service/service_mixins.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 10 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 12 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 13 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 14 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 15 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 16 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/registered_data/registered_data_service_mongodb.py (hunk 17 above)
Comment (Piokor) [reply grisera_api/registered_data/registered_data_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 18 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 20 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 22 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 23 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 24 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 25 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/tests/tests_mongodb/test_registered_channel.py (hunk 50 above)
Comment (Piokor) [reply grisera_api/tests/tests_mongodb/test_registered_channel.py]: fixed

File: grisera_api/tests/tests_mongodb/test_observable_information.py (hunk 48 above)
Comment (Piokor) [reply grisera_api/tests/tests_mongodb/test_observable_information.py]: fixed

File: grisera_api/registered_data/registered_data_service_mongodb.py (hunk 47 above)
Comment (Piokor) [reply grisera_api/registered_data/registered_data_service_mongodb.py]: fixed

File: grisera_api/registered_channel/registered_channel_service_mongodb.py (hunk 45 above)
Comment (Piokor) [reply grisera_api/registered_channel/registered_channel_service_mongodb.py]: fixed

File: grisera_api/registered_channel/registered_channel_service_mongodb.py (hunk 44 above)
Comment (Piokor) [reply grisera_api/registered_channel/registered_channel_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 43 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 42 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 40 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 38 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 36 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/registered_data/registered_data_service_mongodb.py (hunk 35 above)
Comment (Piokor) [reply grisera_api/registered_data/registered_data_service_mongodb.py]: fixed

File: grisera_api/registered_data/registered_data_service_mongodb.py (hunk 34 above)
Comment (Piokor) [reply grisera_api/registered_data/registered_data_service_mongodb.py]: fixed

File: grisera_api/registered_data/registered_data_service_mongodb.py (hunk 33 above)
Comment (Piokor) [reply grisera_api/registered_data/registered_data_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 31 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/recording/recording_service_mongodb.py (hunk 29 above)
Comment (Piokor) [reply grisera_api/recording/recording_service_mongodb.py]: fixed

File: grisera_api/observable_information/observable_information_service_mongodb.py (hunk 28 above)
Comment (Piokor) [reply grisera_api/observable_information/observable_information_service_mongodb.py]: fixed

File: grisera_api/mongo_service/service_mixins.py (hunk 27 above)
Comment (Piokor) [reply grisera_api/mongo_service/service_mixins.py]: fixed

**Review by marcel2012 — APPROVED** [review]

**Review by mrwrob — APPROVED** [review]
LGTM  👍

**Pull request merged**